packages/http-crawler/src/internals/file-download.ts

```diff
@@ -81,16 +73,16 @@ interface ContextInternals {
  * ```
  */
 export class FileDownload extends BasicCrawler<FileDownloadCrawlingContext> {
-    #contextInternals = Symbol('contextInternals');
-
     // TODO hooks
     constructor(options: BasicCrawlerOptions<FileDownloadCrawlingContext> = {}) {
         super({
             ...options,
             contextPipelineBuilder: () =>
                 ContextPipeline.create<CrawlingContext>().compose({
-                    action: this.initiateDownload.bind(this),
-                    cleanup: this.cleanupDownload.bind(this),
+                    action: async (context) => this.initiateDownload(context),
+                    cleanup: async (context) => {
+                        await (context.response.body ? finished(context.response.body as any) : Promise.resolve());
```

Comment:
But what if the request handler doesn't even start reading the stream? Or if it crashes midway? We should make sure we prevent hanging in that case.

Reply:
make it an issue please :smile: